### DISCUSSION TITLE
So many dlls!

### DISCUSSION THREAD

**Discussion by RallyTuning** [dotnet/SqlClient#1984] [Q&A] [answered]
Let's suppose i have an "hello world" app, that just read, insert 1 row, update the row and the delete it. The app is 10kb, single exe. Why i need 23 more dlls?

![image](https://user-images.githubusercontent.com/50233545/230740403-6d02ae42-0a0d-4b04-acc8-24ed84f0d0e0.png)

If i want to make my single dll full of my common classes/functions and a simple connection to a database, i really need my dll + these 23? There is a way to merge them? Considering i'm building a dll, can't these be merged in the man dll?

**Comment by ErikEJ** [comment]
You needed most of these when you used System.Data.SqlClient, they were just installed machine wide.

**Reply by RallyTuning** [reply]
If i remember correct, in VB .Net you just need one reference, the dlls was in the Framework already... so now i need 6,56 MB (23 dlls) for an app that is just 100 KB? Total: 6,57 MB (1% is my code) 😕

**Comment by roji** [comment, marked as answer]
For a lot of these, see #1108.

**Comment by RallyTuning** [comment]
### 🆕 Microsoft.Data.SqlConnect:
**_WinExe app:_**
- 23 different dlls
- File .exe size from 50 KB to 5,50 MB (empty project)

**_Personal DLL library:_**
- Your dll + other 23 different dlls (5+ MB)
---
### 👴🏻 System.Data.SqlConnect:
**_WinExe app:_**
- 2 dlls (sni.dll & System.Data.SqlClient.dll) [around 400 KB total)
- File .exe size: no increases

**_Personal DLL library:_**
- 2 dlls (sni.dll & System.Data.SqlClient.dll) [around 1 MB total)

---
_Sorry but i stick with the old libraries until Microsoft will split mandatory dlls from useless ones. (as me, i think many other will do the same)_

**Reply by ErikEJ** [reply]
@RallyTuning Is that a .net framework app?

**Reply by RallyTuning** [reply]
.Net 6 and .Net 7
